tests/mir_test_framework/window_management_test_harness.cpp

```diff
@@ -0,0 +1,232 @@
+    miral::WindowManagerTools tools;
+};
+
+void mir_test_framework::WindowManagementTestHarness::SetUp()
+{
+    self = std::make_shared<Self>(this);
+    miral::SetWindowManagementPolicy policy([&](miral::WindowManagerTools const& tools)
+    {
+        self->tools = tools;
+        return get_builder()(tools);
+    });
+    policy(server);
+
+
+    auto fake_display = std::make_unique<mtd::FakeDisplay>(get_output_rectangles());
+    self->display = fake_display.get();
+    preset_display(std::move(fake_display));
+
+    mir_test_framework::HeadlessInProcessServer::SetUp();
+}
+
+void mir_test_framework::WindowManagementTestHarness::TearDown()
+{
+    mir_test_framework::HeadlessInProcessServer::TearDown();
+}
+
+auto mir_test_framework::WindowManagementTestHarness::open_application(
+    std::string const& name) -> miral::Application
+{
+    return server.the_shell()->open_session(
+        __LINE__,
+        mir::Fd{mir::Fd::invalid},
+        name,
+        std::shared_ptr<mir::frontend::EventSink>());;
+}
+
+auto mir_test_framework::WindowManagementTestHarness::create_window(
+    miral::Application const& session,
+    mir::shell::SurfaceSpecification spec) -> miral::Window
+{
+    auto stream = std::make_shared<mir::test::doubles::StubBufferStream>();
+    self->streams.push_back(stream);
+    spec.streams = std::vector<msh::StreamSpecification>();
+    spec.streams.value().push_back(msh::StreamSpecification{
+        stream,
+        geom::Displacement{},
+    });
+
+    self->display->configuration()->for_each_output([&](mg::DisplayConfigurationOutput const& output)
+    {
+        spec.output_id = output.id;
+    });
+    auto surface = server.the_shell()->create_surface(
+        session,
+        {},
+        spec,
+        self,
+        &mir::immediate_executor);
+    self->known_surfaces.push_back(surface);
+    server.the_shell()->surface_ready(surface);
+
+    return {session, surface};
+}
+
+void mir_test_framework::WindowManagementTestHarness::publish_event(MirEvent const& event)
+{
+    self->work_queue.enqueue([&]()
+    {
+        server.the_shell()->handle(event);
+    });
+    self->work_queue.run();
+}
+
+void mir_test_framework::WindowManagementTestHarness::request_resize(
+    miral::Window const& window,
+    MirInputEvent const* event,
+    MirResizeEdge edge)
+{
+    self->work_queue.enqueue([&]()
+    {
+        auto surface = window.operator std::shared_ptr<ms::Surface>();
+        server.the_shell()->request_resize(surface->session().lock(), surface, event, edge);
+    });
+    self->work_queue.run();
+}
+
+void mir_test_framework::WindowManagementTestHarness::request_move(
+    miral::Window const& window, MirInputEvent const* event)
+{
+    self->work_queue.enqueue([&]()
+    {
+        auto surface = window.operator std::shared_ptr<ms::Surface>();
+        server.the_shell()->request_move(surface->session().lock(), surface, event);
+    });
+    self->work_queue.run();
```

Comment:
...and likewise, here
```suggestion
    auto surface = window.operator std::shared_ptr<ms::Surface>();
    server.the_shell()->request_move(surface->session().lock(), surface, event);
```